security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -252,9 +260,15 @@ func (sf *SecretFetcher) scrtAdded(obj interface{}) {
 	// If there is root cert secret with the same resource name, delete that secret now.
 	sf.secrets.Delete(rootCertResourceName)
 	if len(newRoot) > 0 {
+		certExpireTime, err := nodeagentutil.ParseCertAndGetExpiryTimestamp(newRoot)
+		if err != nil {
```

Comment:
if ```err != nil```, do we still want to add it to cache and push down, or stop to proceed ?

Reply:
Originally, if operator creates a k8s secret with bad cert/cacert, I think it is up to operator to change the k8s secret. After a second thought, it would be better to enforce at Citadel agent side.

If k8s contains server cert only, and parsing cert fails, skip loading that secret into secretfetcher's cache, and skip updating cachemanager and pushing to Envoy.
If k8s contains server cert and ca cert, if any cert fails to parse, skip loading that secret into secretfetcher's cache, skip updating cache manager and pushing to Envoy.